src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/DB2Engine.java

```diff
@@ -862,8 +900,11 @@ protected ResultIterator createResultIterator(Statement statement, String sql) t
 
             s = conn.createStatement();
             rsColumns = s.executeQuery(
-                String.format("SELECT NAME, COLTYPE,SCALE  FROM sysibm.SYSCOLUMNS WHERE tbname='%s' and TBCREATOR=UPPER('%s')",
-                    name, Optional.ofNullable(properties.getSchema()).orElse(properties.getUsername()))
+                String.format(
+                    "SELECT NAME, COLTYPE, SCALE FROM sysibm.SYSCOLUMNS WHERE tbname LIKE '%s' and TBCREATOR LIKE '%s'",
```

Comment:
The _getMetadata(final String name)_ method was calling
_java.sql.DatabaseMetaData.getColumns(String catalog, String schemaPattern, String tableNamePattern, String columnNamePattern)_
for most DB engines (DB2 was one of the exceptions).

Because of that, the "name" in reality was a table name pattern which might include wildcards.
Since DB2 driver doesn't implement that _getColumns_ method, this one should behave is if it did, by accepting patterns instead of only exact names for schema and table; for this to work, it is necessary to use LIKE.